packages/plugins/response-cache/src/plugin.ts

```diff
@@ -295,11 +296,10 @@ export function useResponseCache<PluginContext extends Record<string, any> = {}>
       if (processedSchemas.has(schema)) {
         return;
       }
-      // Check if the schema has @cacheControl directive
       const cacheControlDirective = schema.getDirective('cacheControl');
-      if (cacheControlDirective) {
```

Comment:
Is this really needed because this visits over the schema even if `cacheControl` is not present.

Reply:
Yeah this is by no means ready to merge, I honestly did not want to risk optimizing a solution that I'm not sure is actually wanted. If we're already talking about it though, what would you suggest? I would probably try to iterate over the schema twice. Keep the handling for `@cacheControl` as it is, and generate the `Map` in a separate iteration separately, depending on `invalidateViaMutations`

Reply:
I think that it's ok like this. If you want to make things more clear, you can eventually create functions with meaningful names to make it clear what is done in each visitor hook. But visiting the entire schema twice can hurt initialisation performance and hurt the startup time :-)

Reply:
And I think we should always build the Map, even when automatic cache invalidation is turned off, since we still want to be able to manually invalidate the cache.